src/db.c

```diff
@@ -81,21 +81,35 @@ int dbIteratorGetCurrentSlot(dbIterator *dbit) {
     return dbit->slot;
 }
 
+/* Used in dbIteratorNext, when getting the next entry in multi slot db, get
+ * the iterator of the next safe dictionary. */
+void dbIteratorInitNextSafeIterator(dbIterator *dbit, dict *d) {
```

Comment:
why should this be a separate function and not part of dbIteratorNext?
the comment says it's specifically for that function, i don't see a reason to use it anywhere else, and i don't think it's complicated enough to justify it's own function.

do you aggree?

Reply:
i though it would be nice to have a function to init the next iterator, and it can hide prev dict details.

i also agree with your point, if you have a strong wish, i can remove this function

Reply:
I think this should be part of `dbIteratorNext`, not a separate function. `dbIterator`->`dictIterator`->`dict` it is a long chain, hiding details may actually introduce risks.

Reply:
make sense, i am doing it